include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1167 @@
+ * Free Software Foundation with the addition of the following permission added
+ * to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+ * IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+ * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+/**
+ * Implements Google Calendar Syncing
+ *
+ * @license https://raw.githubusercontent.com/salesagility/SuiteCRM/master/LICENSE.txt
+ * GNU Affero General Public License version 3
+ * @author Benjamin Long <ben@offsite.guru>
+ */
+
+class GoogleSync
+{
+
+    /** @var array An array of user id's we are going to sync for */
+    public $users = array();
+
+    /** @var User The SuiteCRM User Bean we're currently working with */
+    public $workingUser;
+
+    /** @var \Google_Client The Google client object for the current sync job */
+    public $gClient;
+
+    /** @var \Google_Service_Calendar The Google Calendar Service Object */
+    public $gService;
+
+    /** @var array The Google AuthcConfig json */
+    public $authJson = array();
+
+    /** @var string The local timezone */
+    public $timezone;
+
+    /** @var string The Calendar ID */
+    public $calendarId;
+
+    /** @var array An array of SuiteCRM meeting id's that we've already synced this session */
+    public $syncedList = array();
+
+    /** @var object A Database Instance */
+    private $db;
+
+    /**
+     * Class Constructor
+     */
+    public function __construct()
+    {
+        if (isset($_SERVER['GSYNC_LOGLEVEL'])) {
+            LoggerManager::getLogger()->setLevel($_SERVER['GSYNC_LOGLEVEL']);
```

Comment:
Should I take this out of the code altogether? It's for debugging just this class when running from cron.php. I'm not sure how to set the debug level back when its set in the constructor. As it functions now, using this will output debug/info/error/etc level logging for everything in or run from this class. It's useful when I'm trying to figure out what's broken, but I'm not sure it would be useful to anyone else.

Reply:
An example of how this works from the command line:
$ GSYNC_LOGLEVEL=debug php -f cron.php

Reply:
Done